include/revng/Model/Binary.h

```diff
@@ -282,15 +283,51 @@ class model::Binary : public model::generated::Binary {
   bool verifyTypeDefinitions(bool Assert) const debug_function;
   bool verifyTypeDefinitions() const debug_function;
 
-  bool verifyGlobalNamespace(VerifyHelper &VH) const;
-  bool verifyGlobalNamespace(bool Assert) const debug_function;
-  bool verifyGlobalNamespace() const debug_function;
-
 public:
   void dumpTypeGraph(const char *Path) const debug_function;
 
 public:
   MetaAddressRangeSet executableRanges() const;
+
+private:
+  // It's defined here instead of the TTG because we don't really want it to
+  // ever get serialized.
+  mutable std::optional<model::TypeOfGlobalNameCache> TheGlobalNameCache = {};
```

Comment:
Oh no. Not `mutable`.

I'm starting to think that all the naming business should be an external object that inspects the model.
There's no real reason to embed it in the model.

If you want the names, you create your object, you take care of its lifetime and make sure you don't change the model while such object is alive.

```c++
class Namer {
private:
  GlobalNamespace;
public:
  Namer(const model::Binary &Binary) { /* Initialize GlobalNamespace */ }

public:
  std::string getArgumentName(...);
};

Reply:
Please take a look.